crates/ruff_linter/src/rules/flake8_errmsg/snapshots/ruff_linter__rules__flake8_errmsg__tests__custom.snap

```diff
@@ -177,4 +177,86 @@ EM.py:60:35: EM101 Exception must not use a string literal, assign to variable f
    |
    = help: Assign to variable; remove string literal
 
+EM.py:64:24: EM102 [*] Exception must not use an f-string literal, assign to variable first
+   |
+63 | def f_triple_quoted_string():
+64 |     raise RuntimeError(f"""This is an {"example"} exception""")
+   |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ EM102
+   |
+   = help: Assign to variable; remove f-string literal
+
+ℹ Unsafe fix
+61 61 | 
+62 62 | 
+63 63 | def f_triple_quoted_string():
+64    |-    raise RuntimeError(f"""This is an {"example"} exception""")
+   64 |+    msg = f"""This is an {"example"} exception"""
+   65 |+    raise RuntimeError(msg)
+65 66 | 
+66 67 | 
+67 68 | # Generate a violaiton for this, but don't fix it
+
+EM.py:77:9: EM103 [*] Exception must not use a `.format()` string directly, assign to variable first
+   |
+75 |   def f_multi_line_string2():
+76 |       raise RuntimeError(
+77 |           "This is an {example} exception".format(
+   |  _________^
+78 | |             example="example"
+79 | |         )
+   | |_________^ EM103
+80 |       )
+   |
+   = help: Assign to variable; remove `.format()` string
+
+ℹ Unsafe fix
+73 73 | 
+74 74 | 
+75 75 | def f_multi_line_string2():
+76    |-    raise RuntimeError(
+   76 |+    msg = (
```

Comment:
So these probably _should_ be omitted, right? But we don't have a general way to know if the inner expression needs parentheses, right?

Reply:
Yes, that's correct.